tests/unit/wasm-c-api/enhanced_gen_wasm_memory_new_internal_test.cc

```diff
@@ -0,0 +1,547 @@
+/*
+ * Copyright (C) 2019 Intel Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <gtest/gtest.h>
+#include <cstring>
+
+#include "bh_platform.h"
+#include "wasm_c_api.h"
+#include "wasm_c_api_internal.h"
+#include "wasm_runtime_common.h"
+
+#ifndef own
+#define own
+#endif
+
+// Simple WebAssembly module with memory (1 page minimum, 10 pages maximum)
+static const uint8_t wasm_memory_module[] = {
+    0x00, 0x61, 0x73, 0x6d, // WASM_MAGIC
+    0x01, 0x00, 0x00, 0x00, // WASM_VERSION
+
+    // Type section
+    0x01, // section id: type
+    0x04, // section size
+    0x01, // type count
+    0x60, // function type
+    0x00, // param count
+    0x00, // result count
+
+    // Function section
+    0x03, // section id: function
+    0x02, // section size
+    0x01, // function count
+    0x00, // function 0 type index
+
+    // Memory section
+    0x05, // section id: memory
+    0x03, // section size
+    0x01, // memory count
+    0x00, // memory limits: no maximum
+    0x01, // initial pages: 1
+
+    // Code section
+    0x0a, // section id: code
+    0x04, // section size
+    0x01, // function count
+    0x02, // function 0 body size
+    0x00, // local count
+    0x0b  // end
+};
+
+// WebAssembly module with no memory
+static const uint8_t wasm_no_memory_module[] = {
+    0x00, 0x61, 0x73, 0x6d, // WASM_MAGIC
+    0x01, 0x00, 0x00, 0x00, // WASM_VERSION
+
+    // Type section
+    0x01, // section id: type
+    0x04, // section size
+    0x01, // type count
+    0x60, // function type
+    0x00, // param count
+    0x00, // result count
+
+    // Function section
+    0x03, // section id: function
+    0x02, // section size
+    0x01, // function count
+    0x00, // function 0 type index
+
+    // Code section
+    0x0a, // section id: code
+    0x04, // section size
+    0x01, // function count
+    0x02, // function 0 body size
+    0x00, // local count
+    0x0b  // end
+};
+
+class WasmMemoryNewInternalTest : public ::testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        // Initialize WAMR runtime
+        RuntimeInitArgs init_args;
+        memset(&init_args, 0, sizeof(RuntimeInitArgs));
+        init_args.mem_alloc_type = Alloc_With_System_Allocator;
+
+        ASSERT_TRUE(wasm_runtime_full_init(&init_args));
```

Comment:
Similar to the above, when calling `wasm_engine_new()`, `wasm_runtime_full_init()` should not be executed.